### DISCUSSION TITLE
Peaks at summed time statistics

### DISCUSSION THREAD

**Discussion by Cully81** [krahabb/meross_lan#263] [Q&A] [answered]
Hello,

I would like to have the runtime of my mts200 underfloor heating displayed per day. I have already created a sensor for this, which actually works:

`
  - platform: template
    sensors:
      heizung_esszimmer_on_off:
        value_template: "{{ state_attr('climate.smart_thermostat_123','hvac_action')}}"
  - platform: history_stats
    name: Heizung Esszimmer Laufzeit
    entity_id: sensor.heizung_esszimmer_on_off
    state: "heating"
    type: time
    start: "{{ now().replace(hour=0, minute=0) }}"
    end: "{{ now() }}"
`

However, when I look at the graph, I see peaks that should not be there. The running time is about 2 hours every day. The peaks then sometimes go up to 10 hours, although the heating is not on. The graph then also falls back again within the day. At some point, the sensor is apparently wrongly processed. But I do not know where. Can anyone help?

![grafik](https://user-images.githubusercontent.com/96849812/221607412-0bbf7899-db64-45b4-8f5e-51bc65206653.png)

Is the error in my code? Or what could be the reason?

Thanks
Loki

**Comment by krahabb** [comment, marked as answer]
I've just looked at the docs for history_stats and the example says you should put 
`start: "{{ now().replace(hour=0, minute=0, second=0) }}"` to reset at the start of the day. I'd even say it would be better to also reset the microseconds part of the date and I guess this would be needed to avoid 'strange' behaviors when the stats get calculated by inspecting the recorder database for the entity. Maybe the implementation is not so straightforward as it could seem so I'd just try to be more strict on this (if you didn't already)

**Comment by Cully81** [comment]
Hi @krahabb 

Thank you very much for your answer.
That was indeed the solution. I simply added the seconds.

`    start: "{{ now().replace(hour=0, minute=0, second=0) }}"`

And there you go, it looks good. 

![grafik](https://user-images.githubusercontent.com/96849812/222950225-fb3aebdb-6b4c-4be0-bd0b-188f70685a52.png)
